### PR TITLE
Fix/add -- Code better follows MVSC pattern and addition of repository pattern

### PR DESCRIPTION
**Pull request by AndreasBM1**
# Description
The code now follows the actual structure of the MVSC pattern, where the code only calls the service which in turn calls the model.

The repository pattern has also been added as part of the model layer, as of now it is a partial integration to show the idea, a more thorough version is likely to spawn.

## What is added/changed/removed
Some files for repositories have been added,
Changed where code calls the model directly instead of the service.

## Testing
Runit, beit, sniffit

# Checks
- [ ] Updated / added tests for these changes
- [x] PHPunit locally passed
- [x] Codesniffer locally passed
- [x] Behat locally passed

### CODE REVIEW

**Review by DeeKahy — APPROVED** [review]
may the force be with you

File: server/moodle/mod/livequiz/classes/models/question.php [hunk 1]
```diff
@@ -373,16 +292,24 @@ public function prepare_for_template(stdClass $data): stdClass {
...
+     * @throws dml_exception
      */
-    public function is_answer_in_question($answerid): bool {
-        foreach ($this->answers as $answer) {
-            if ($answer->get_id() == $answerid) {
-                return true;
-            }
-        }
-        return false;
+    public static function get_question_with_answers_from_id($id): question {
```
Comment (Laczse) [inline server/moodle/mod/livequiz/classes/models/question.php:301]: Denne method skal vel være i question_repository.php?

File: server/moodle/mod/livequiz/classes/services/livequiz_services.php [hunk 2]
```diff
@@ -203,6 +191,7 @@ private function insert_question_with_relations(question $question, int $lecture
      * @throws Exception
      */
     private function submit_answers(int $questionid, array $answers): void {
+        $answerrepository = new answer_repository();
```
Comment (Laczse) [inline server/moodle/mod/livequiz/classes/services/livequiz_services.php:194]: Maybe make this into an instance variable as it is used in multiple places and the object is not changed. Maybe do the same for the other repositories

File: server/moodle/mod/livequiz/classes/repositories/question_repository.php [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    /**
+     * Updates a question in the database.
+     *
+     * @param question $question
+     * @return void
+     * @throws dml_exception
+     */
+    public function update_question(question $question): void {
```
Comment (Laczse) [inline server/moodle/mod/livequiz/classes/repositories/question_repository.php:114]: Hvorfor er denne ikke static, når alle andre methods i classen er`?

**Review by Laczse — APPROVED** [review]
Looks good. Have some comments but nothing that is required for approving this PR

File: server/moodle/mod/livequiz/classes/repositories/question_repository.php (hunk 3 above)
Comment (AndreasBM1) [reply server/moodle/mod/livequiz/classes/repositories/question_repository.php:114]: Jeg tager kommentarene med til næste PR

**Review by ChristofferGamel — APPROVED** [review]

**Pull request merged**